### PR TITLE
Unified Build Scripts

### PR DESCRIPTION
**Pull request by anurse**
Ok, I've got most of the Windows stuff working, <del>still need to port @eerhardt 's package cache clearing code (working on that now)</del> (Done!) and <del>get the OS X and Linux builds wired up</del> (Done!)

TODO:

* [x] Fix Mac
* [x] Fix Ubuntu
* [x] Fix CentOS
* [x] Relocate NUGET_PACKAGES to a folder outside the repo root when doing CI build.

/cc @piotrpMSFT @brthor @Sridhar-MS @eerhardt @pakrym @davidfowl

### CODE REVIEW

File: scripts/Microsoft.DotNet.Cli.Build.Framework/project.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "version": "1.0.0-*",
+
+    "dependencies": {
+        "NETStandard.Library": "1.0.0-rc2-23805",
+        "System.Diagnostics.Process": "4.1.0-rc2-23805"
```
Comment (anurse) [inline scripts/Microsoft.DotNet.Cli.Build.Framework/project.json]: Was Process intentionally removed from the standard library? That seems like a pretty "standard library" kind of thing to me...

/cc @ericstj @davidfowl

Comment (ericstj) [inline scripts/Microsoft.DotNet.Cli.Build.Framework/project.json]: Yes, @KrzysztofCwalina is going to replace it with a new Process API but didn't want the current windows-centric API to be forever legacy on new platforms.  @KrzysztofCwalina do you have an issue tracking the new API?

Comment (anurse) [inline scripts/Microsoft.DotNet.Cli.Build.Framework/project.json]: Ooooh, fancy. That's seems reasonable to me. Thanks for the update! :)

**Comment by anurse** [comment]
@dotnet-bot test CentOS7.1 Debug Build please
@dotnet-bot test Ubuntu Debug Build please
@dotnet-bot test Windows_NT Release Build please

**Comment by anurse** [comment]
@dotnet-bot test this please

**Comment by anurse** [comment]
@dotnet-bot test OSX Debug Build please
@dotnet-bot test Ubuntu Debug Build please
@dotnet-bot test Ubuntu Release Build please

**Comment by anurse** [comment]
@dotnet-bot test this please

**Comment by anurse** [comment]
@dotnet-bot test Ubuntu Debug Build please
@dotnet-bot test Ubuntu Release Build please
@dotnet-bot test CentOS7.1 Debug Build please
@dotnet-bot test CentOS7.1 Release Build please

**Comment by anurse** [comment]
@dotnet-bot test Windows_NT Release Build please

**Comment by anurse** [comment]
@dotnet-bot test Windows_NT Release Build please

**Comment by anurse** [comment]
FYI: Do not merge, even if all looks green. It appears there is a bug in some of the error reporting.

**Comment by davidfowl** [comment]
!!!!

**Comment by anurse** [comment]
Don't get too excited:

> 21:54:35 [TARGET     <] [FAIL] [2016-02-12T05:54:38.7362590+00:00] Default                      (Standard Goals)
> 21:54:35 Build failed: Linux not working yet

That's not a great thing :). It's not reporting the exit code right.

**Comment by pakrym** [comment]
It would be nice if someone installed https://wiki.jenkins-ci.org/display/JENKINS/AnsiColor+Plugin for cli builds

**Comment by anurse** [comment]
@dotnet-bot test CentOS7.1 Release Build please

**Comment by anurse** [comment]
This is real folks. 8 successful checks on the unified build scripts! Ready to merge!

/cc @piotrpMSFT @Sridhar-MS @brthor @pakrym @davidfowl @livarcocc

**Comment by anurse** [comment]
@dotnet-bot test this please

**Comment by anurse** [comment]
@dotnet-bot test OSX Debug Build please

**Comment by anurse** [comment]
I've successfully verified the builds work on VSO as well as Jenkins now.

**Comment by anurse** [comment]
@dotnet-bot test OSX Debug Build please
@dotnet-bot test OSX Release Build please
@dotnet-bot test Ubuntu Debug Build please
@dotnet-bot test Ubuntu Release Build please

**Comment by anurse** [comment]
@dotnet-bot test OSX Debug Build please
@dotnet-bot test OSX Release Build please
@dotnet-bot test Ubuntu Debug Build please
@dotnet-bot test Ubuntu Release Build please

**Pull request merged**